### ISSUE TITLE
Some RealmChangeResults features that i think would be nice to have :)

### ISSUE THREAD

**Issue by hay12396** [realm/realm-java#2747]
Hello,

In case i am registering a change listener to RealmResults, and one of the RealmObject in that RealmResults was changed, i am getting back the RealmResults via the onChange method.

Because i want to have the spesific RealmObject that was changed, i am now registering a changelistener to each RealmObject in the RealmResults, so that the onChange method will give me the RealmObject that i need.

***** I was thinkin about adding something like "addChildChangeListener" method that will add a change listener to each of the RealmObject that collection contains, so one would not need to loop through the collection to perform this action.

Now let's say that my list is getting updated every period of time, i will than need to going through my list and register a changelistener to the new items in my list.

***** But in order to do that it would be really nice if we had something like a "hasChangeListener" or "getChangeListeners" so we could check if the specific object already have a registered change listener(so we won't register another one).

***** In addition, once i'm done with the specific list (say the user close the fragment that was holding that list) i need to loop through my list and unregister the listeners.
So my second idea was to have some kind of "removeAllChildListeners" method on the Realm collection api (RealmResults/RealmCollection/RealmList) that will remove the change listeners of the containing RealmObjects.

In conclusion, in my opinion the following methods would be really helpfull:

1. getChangeListeners/hasChangeListener : a method that will return the list of listeners registered to that object/ returns a boolean wether that object has change listeners registered to him.

2. addChildChangeListener : a method that will add a change listener to each of the RealmObjects in that collection.

3. removeChildChangeListeners : a method that will remove all the change listeners of the RealmObjects in that collection.

With regards,
Hay.

**Comment by cmelchior** [comment]
Hi @hay12396 
Thanks for the detailed feedback, and we agree that collection listeners would be useful. This is already being tracked in #989 so i'll close this as an duplicate, but you can follow progress there.
